### DISCUSSION TITLE
Performance Android Demo Gallery

### DISCUSSION THREAD

**Discussion by AlleSchonWeg** [unoplatform/uno#15609] [General]
Hi,
i tested the demo project with my Honor 9 phone: https://play.google.com/store/apps/details?id=com.nventive.uno.ui.demo. But the performance is not so good.
Scrolling is lagging and not smooth. In the flyoutmenu and in content area. Also the clicks on the expand elements in the flyout menu feels not smooth. There is a little delay before the menu expands.
My XF apps with much more content on the same phone has no such scrolling lags.
Is this the "expected" performance for a uno android app?
Thank you!

**Comment by sasakrsmanovic** [comment]
Hello; thanks for considering Uno Platform for your mobile project. 

On a real, performance optimized app you will experience much different performance. To experience it I invite you to try some of the real customer apps that are running in the app store. i.e.
https://play.google.com/store/apps/details?id=co.tradezero.zeromobile
https://play.google.com/store/apps/details/Jean_Coutu?id=com.jeancoutu.pjc
https://play.google.com/store/apps/details?id=ca.ottawa.cityofottawa 

The Gallery app you are pointing to is not a typical app - its purpose is to showcase as many controls that are at your disposal for a mobile app. 

Your point is fair though, we should work on optimizing it as well, or providing a different sample that resembles the real app in a better way. 

What XF apps do you have; I'd love to take a look.

**Reply by AlleSchonWeg** [reply]
Hi,
thanks for your reply. Unfortunately i can't install the first two mentioned apps on my phone:
> Diese App ist für dein Gerät nicht verfügbar

I have a honor 9 Google Phh-Treble With GApps. Android 12. I can install the tradezero app but an account is needed. 

The XF app is only available for our customers. The app shows there private data in charts and tables. It is a simple app for android and iOS. Just an nice to have AddOn for our customers.
We have a TabbedPage which contains up to 4 NavigationPages. The root pages are menus(ListViews and CollectionViews) where the user can select different charts (custom skia controls) or tables (combination of grids, scrollviews and labels). On almost every page we have one or more animations.
I started migration to MAUI but almost nothing works. Especially animations crashes the app after a few minutes. Then problems with Borders, Shadows, Ripple Effects, Systemfonts etc. We decided to suspend migration, because we can't release an app to our customers with such errors/problems. Now we evaluate alternatives. Thats the reason why i'm here :)

If we go with uno we need the toolkit, because of the TabBar. Must we buy a licence if revenue is larger than 1.000.000 $? (https://platform.uno/uno-toolkit/)? Or is this the MIT license: https://github.com/unoplatform/uno.toolkit.ui?tab=MIT-1-ov-file#readme ?

Thanks again

**Reply by sasakrsmanovic** [reply]
Sorry to hear about your MAUI migration challenges. We have some documentation to help with XF to Uno Platform migration https://platform.uno/xamarin-forms/

Please reach out to me at sasha at platform dot uno to grant you a license for the toolkit you are asking for.
